marxs/optics/multiLayerMirror.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+from astropy.io import ascii
+
+from .base import FlatOpticalElement
+from ..math.pluecker import *
+
+
+class MultiLayerMirror(FlatOpticalElement):
+    '''Multilayer mirror with varying layer thickness along one axis
+    
+    The distance between layers changes along the local y axis.
+    All reflectivity data is assumed to be for a single, desired angle. There
+    is currently no way to enter varying reflection that depends on the angle
+    of incidence.
+    Provide reflectivity data in a file with columns:
+    	'X(mm)' - position along the "changing" axis
+    	'Peak lambda' - wavelength with maximum reflection at a given position
+    	'Peak' - maximum reflection at a given position
+    	'FWHM(nm)' - full width half max, measure of width of reflection Gaussian peaks
+    
+    Parameters
+    ----------
+    fileCode: string
+    	serial number of mirror, used to open reflection data file
+    '''
+    def __init__(self, fileCode, **kwargs):
+        self.fileName = './marxs/optics/docs/' + fileCode + '.txt'
+        if ('zoom' not in kwargs):
+        	kwargs['zoom'] = np.array([1, 24.5, 12])   # in mm
```

Comment:
An arbitrary zoom is fine as well. This is the size of all of the mirrors I know of, so I wanted to include it as a default. Only one dimension can be taken from the file, so I figured if you wanted to use a mirror of a different size you could just specify zoom and override the default. I guess this might only be helpful for Herman, because other labs could have different "typical" mirror sizes, but it would be nice to not have to reenter zoom repeatedly for Herman's simulation. Do you think I should take it out?